apps/mobile/src/components/GalleryEditor/SortableRow.tsx

```diff
@@ -0,0 +1,273 @@
+import { FlashList } from '@shopify/flash-list';
+import { useCallback } from 'react';
+import { useWindowDimensions } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import { trigger } from 'react-native-haptic-feedback';
+import Animated, {
+  AnimatedRef,
+  runOnJS,
+  scrollTo,
+  SharedValue,
+  useAnimatedReaction,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+import { ListItemType } from './GalleryEditorRender';
+import { ItemHeights, Positions } from './SortableRowList';
+
+type Props = {
+  children: React.ReactNode;
+  index: number;
+  positions: SharedValue<Positions>;
+  animatedIndex: SharedValue<number | null>;
+  itemHeights: SharedValue<ItemHeights>;
+  scrollContentOffsetY: SharedValue<number>;
+  scrollViewRef: AnimatedRef<FlashList<ListItemType>>;
+
+  onDragEnd: (data: string[]) => void;
+};
+
+export function SortableRow({
+  children,
+  index,
+  positions,
+  animatedIndex,
+  itemHeights,
+  scrollContentOffsetY,
+  scrollViewRef,
+  onDragEnd,
+}: Props) {
+  const itemHeight = itemHeights.value[index] ?? 0;
+  // Shared values for gesture handling
+  const contextY = useSharedValue(0);
+  const translateX = useSharedValue(0);
+
+  // Shared values for tracking gesture and animation state
+  const wasLastActiveIndex = useSharedValue(false);
+
+  // Animated reaction to update last active index
+  useAnimatedReaction(
+    () => animatedIndex.value,
+    (currentActiveIndex) => {
+      if (currentActiveIndex) {
+        wasLastActiveIndex.value = currentActiveIndex === index;
+      }
+    }
+  );
+
+  // Derived value to check if the gesture is active
+  const isGestureActive = useDerivedValue(() => {
+    return animatedIndex.value === index;
+  }, [index]);
+
+  // Callback to get the position of a list item
+  const getPosition = useCallback(
+    (itemIndex: number) => {
+      'worklet';
```

Comment:
wtf is `worklet`? lol

Reply:
Basically its a way to tell reanimated to run the function in the UI thread instead of JS thread. Useful for animation purpose. 

Reference: https://docs.swmansion.com/react-native-reanimated/docs/2.x/fundamentals/worklets/